proposals/913-components-repo/README.md

```diff
@@ -0,0 +1,497 @@
+- [Risks and Mitigations](#risks-and-mitigations)
+- [Test Plan](#test-plan)
+  - [Unit & Compile Checks](#unit--compile-checks)
+  - [Continuous Integration](#continuous-integration)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Separate Repositories](#separate-repositories)
+  - [Keep the Existing `components` Directory in the KFP Repo](#keep-the-existing-components-directory-in-the-kfp-repo)
+  <!-- /toc -->
+
+## Summary
+
+Establish a dedicated Kubeflow Pipelines (KFP) repository\* that hosts reusable components and full pipelines under a
+consistent structure, governance policy, and release cadence. The repository will package officially supported assets as
+a Python distribution for easy consumption, while providing a parallel space for vetted third-party contributions. The
+project introduces standardized metadata, documentation, testing, and maintenance automation to make components
+discoverable, reliable, and safe to adopt.
+
+\*Working title `kubeflow/pipelines-components`; the final repository name will be confirmed during implementation.
+
+## Motivation
+
+Kubeflow Pipelines currently ships sample components in-tree under `components/`, making reuse and contribution
+difficult because:
+
+- Assets are versioned with the main KFP repo, so release timing is tied to the entire project.
+- Documentation is inconsistent and buried.
+- There is no uniform metadata that tells users about compatibility, dependencies, or support status.
+- CI coverage is uneven, so users do not know if a component still works.
+- Third-party components mix with community-maintained artifacts without clear ownership or expectations.
+- Components are not published as a Python package today, so consumption requires cloning the monorepo or copying
+  snippets manually.
+
+A dedicated repository with purpose-built tooling will make it easier for users to discover, evaluate, and integrate KFP
+assets while allowing the community to maintain them at its own cadence.
+
+This catalog also creates a durable bridge between Kubeflow Pipelines and the broader Kubeflow ecosystem: it gives the
+SDK and API server a canonical source of reusable building blocks, provides SIGs with a venue to showcase
+Kubernetes-native MLOps patterns, and offers users a distribution channel that evolves independently from the core KFP
+release train. Centralizing community-supported assets in one place keeps the SDK surface area lean while encouraging
+experimentation in third-party areas that can later graduate into core.
+
+### Goals
+
+1. Move reusable components and pipelines into a dedicated GitHub repository with clear structure and governance.
+2. Provide standardized metadata, documentation, and testing requirements for every asset.
+3. Ship an installable Python package for core (community-maintained) components that is versioned to match Kubeflow
+   releases.
```

Comment:
Given that core components probably be part of Kubeflow SDK ? So we can something like this:
cc @kubeflow/kubeflow-sdk-team 
```suggestion
3. Ship an installable Python package for core (community-maintained) components that is versioned to match Kubeflow SDK releases.
```